src/wp-includes/class-wp-scripts.php

```diff
@@ -714,6 +781,262 @@ public function in_default_dir( $src ) {
 		return false;
 	}
 
+	/**
+	 * This overrides the add_data method from WP_Dependencies, to support normalizing of $args.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle Name of the item. Should be unique.
+	 * @param string $key    The data key.
+	 * @param mixed  $value  The data value.
+	 * @return bool True on success, false on failure.
+	 */
+	public function add_data( $handle, $key, $value ) {
+		if ( 'script_args' === $key ) {
+			$args = $this->get_normalized_script_args( $handle, $value );
+			if ( $args['in_footer'] ) {
+				parent::add_data( $handle, 'group', 1 );
+			}
+			return parent::add_data( $handle, $key, $args );
+		}
+		return parent::add_data( $handle, $key, $value );
+	}
+
+	/**
+	 * Checks all handles for any delayed inline scripts.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @return bool True if the inline script present, otherwise false.
+	 */
+	public function has_delayed_inline_script() {
+		foreach ( $this->registered as $handle => $script ) {
+			// Non standalone scripts in the after position, of type async or defer, are usually delayed.
+			$strategy = $this->get_intended_strategy( $handle );
+			if ( $this->is_non_blocking_strategy( $strategy ) &&
+				$this->has_non_standalone_inline_script( $handle, 'after' )
+			) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Normalize the data inside the $args parameter and support backward compatibility.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string        $handle Name of the script.
+	 * @param array         $args     {
+	 *      Optional. Additional script arguments. Default empty array.
+	 *
+	 *      @type boolean   $in_footer    Optional. Default false.
+	 *      @type string    $strategy     Optional. Values blocking|defer|async. Default 'blocking'.
```

Comment:
```suggestion
	 *      @type bool      $in_footer    Optional. Whether to enqueue the script before `</body>` instead of in the `<head>`.
	 *                                    Default false.
	 *      @type string    $strategy     Optional. The loading strategy.
	 *                                    Accepts 'blocking', 'defer', or 'async'. Default 'blocking'.
```
1. Core uses `bool` rather than `boolean` for type documentation.
2. Added descriptions, and some reformatting.

Reply:
@costdev noting that with https://github.com/10up/wordpress-develop/pull/64 merging shortly, this method will be removed altogether.